src/test/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdStateUpdateServiceTest.java

```diff
@@ -0,0 +1,256 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingResponse;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.ListingStatus;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.RequestDetails;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.State.DORMANT_APPEAL_STATE;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.State.UNKNOWN;
+
+@ExtendWith(MockitoExtension.class)
+class CcdStateUpdateServiceTest {
+
+    @InjectMocks
+    private CcdStateUpdateService underTest;
+
+    @DisplayName("When valid listing Status and list assist case status is given, "
+            + "updateCancelled updates the case data correctly")
+    @ParameterizedTest
+    @CsvSource({
+        "FIXED,LISTED,HEARING",
+        "FIXED,AWAITING_LISTING,READY_TO_LIST",
+    })
+    void testUpdateListed(ListingStatus listingStatus, ListingCaseStatus listingCaseStatus, State expected) throws UpdateCaseException {
+        // given
+        HearingGetResponse hearingGetResponse = HearingGetResponse.builder()
```

Comment:
optional: consider extracting this builder in builder to a fixture method and pass in the values - as it used in most of these tests